### PR TITLE
Fixes several broken python packages for python-unstable

### PR DESCRIPTION
**Pull request by costrouc**
###### Motivation for this change

https://github.com/NixOS/nixpkgs/pull/55757

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by dotlambda** [comment]
"depreciated" -> "deprecated"

File: pkgs/development/python-modules/pint/default.nix [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
+{ lib
 , buildPythonPackage
 , fetchPypi
+, isPy27
+, funcsigs
 }:
 
 buildPythonPackage rec {
-  pname = "pint";
+  pname = "Pint";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pint/default.nix]:
```suggestion
  pname = "pint";
```
Put the line inside `fetchPypi` instead.

File: pkgs/development/python-modules/pynacl/default.nix [hunk 2]
```diff
@@ -1,21 +1,34 @@
...
+, pytest
+, libsodium
+, cffi
+, six
+, hypothesis
+}:
 
 buildPythonPackage rec {
-  pname = "pynacl";
+  pname = "PyNaCl";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pynacl/default.nix]: same here

**Comment by dotlambda** [comment]
@GrahamcOfBorg eval

File: pkgs/development/python-modules/shapely/default.nix [hunk 3]
```diff
@@ -22,11 +22,16 @@ buildPythonPackage rec {
...
 
-  patchPhase = let
+  postPatch =  let
     libc = if stdenv.isDarwin then "libc.dylib" else "libc.so.6";
   in ''
-    sed -i "s|_lgeos = load_dll('geos_c', fallbacks=.*)|_lgeos = load_dll('geos_c', fallbacks=['${geos}/lib/libgeos_c${stdenv.hostPlatform.extensions.sharedLibrary}'])|" shapely/geos.py
-    sed -i "s|free = load_dll('c').free|free = load_dll('c', fallbacks=['${stdenv.cc.libc}/lib/${libc}']).free|" shapely/geos.py
+    substituteInPlace shapely/geos.py \
+      --replace "_lgeos = load_dll('geos_c', fallbacks=alt_paths)" "_lgeos = load_dll('geos_c', fallbacks=['${geos}/lib/libgeos_c${stdenv.hostPlatform.extensions.sharedLibrary}'])" \
+      --replace "free = load_dll('c').free" "free = load_dll('c', fallbacks=['${stdenv.cc.libc}/lib/${libc}']).free"
```
Comment (dotlambda) [inline pkgs/development/python-modules/shapely/default.nix]: I'd prefer doing this with a patch and `substituteAll` so we can easily detect when it no longer applies. Also, no need to use `fallback` when we know the exact file.

File: pkgs/development/python-modules/shapely/default.nix [hunk 4]
```diff
@@ -22,11 +22,16 @@ buildPythonPackage rec {
...
+  '';
+
+  preBuild = ''
+    export GEOS_LIBRARY_PATH="${geos}/lib/libgeos_c${stdenv.hostPlatform.extensions.sharedLibrary}"
```
Comment (dotlambda) [inline pkgs/development/python-modules/shapely/default.nix]: Why is this needed in addition to the patch?

File: pkgs/development/python-modules/gidgethub/default.nix [hunk 5]
```diff
@@ -15,24 +15,28 @@
...
     inherit pname version;
-    sha256 = "1ebe79cf80ad64cb78c880efc7f30ac664e18b80dfd18ee201bf8685cf029628";
+    sha256 = "52119435ba73ddd5e697dae7bec8b93a048bc738720b81691ebd4b4d81d2d762";
   };
 
   buildInputs = [ setuptools pytestrunner ];
   checkInputs = [ pytest pytest-asyncio twisted treq tornado aiohttp ];
   propagatedBuildInputs = [ uritemplate ];
 
+  postConfigure = ''
```
Comment (dotlambda) [inline pkgs/development/python-modules/gidgethub/default.nix]:
```suggestion
  postPatch = ''
```

File: pkgs/development/python-modules/gidgethub/default.nix [hunk 6]
```diff
@@ -15,24 +15,28 @@
...
 
   disabled = pythonOlder "3.6";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "1ebe79cf80ad64cb78c880efc7f30ac664e18b80dfd18ee201bf8685cf029628";
+    sha256 = "52119435ba73ddd5e697dae7bec8b93a048bc738720b81691ebd4b4d81d2d762";
   };
 
   buildInputs = [ setuptools pytestrunner ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/gidgethub/default.nix]:
```suggestion
  nativeBuildInputs = [ setuptools pytestrunner ];
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pint/default.nix (hunk 1 above)
Comment (costrouc) [reply pkgs/development/python-modules/pint/default.nix]: fixed.

File: pkgs/development/python-modules/pynacl/default.nix (hunk 2 above)
Comment (costrouc) [reply pkgs/development/python-modules/pynacl/default.nix]: fixed.

File: pkgs/development/python-modules/gidgethub/default.nix (hunk 5 above)
Comment (costrouc) [reply pkgs/development/python-modules/gidgethub/default.nix]: fixed.

File: pkgs/development/python-modules/gidgethub/default.nix (hunk 6 above)
Comment (costrouc) [reply pkgs/development/python-modules/gidgethub/default.nix]: fixed.

File: pkgs/development/python-modules/shapely/default.nix (hunk 3 above)
Comment (costrouc) [reply pkgs/development/python-modules/shapely/default.nix]: I got rid of using load_dll and fallback.

Comment (costrouc) [reply pkgs/development/python-modules/shapely/default.nix]: Fixed.

File: pkgs/development/python-modules/shapely/default.nix (hunk 4 above)
Comment (costrouc) [reply pkgs/development/python-modules/shapely/default.nix]: fixed. found a way to do in the path phase

File: pkgs/development/python-modules/pynacl/default.nix [hunk 7]
```diff
@@ -1,21 +1,35 @@
...
-    rev = version;
-    sha256 = "0ac00d5bfdmz1x428h2scq5b34llp61yhxradl94qjwz7ikqv052";
+  src = fetchPypi {
+    inherit version;
+    pname = "PyNaCl";
+    sha256 = "0c6100edd16fefd1557da078c7a31e7b7d7a52ce39fdca2bec29d4f7b6e7600c";
   };
 
   checkInputs = [ pytest hypothesis ];
   propagatedBuildInputs = [ libsodium cffi six ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/pynacl/default.nix]: Libsodium should actually not be propagated but I can do that separately.

File: pkgs/development/python-modules/shapely/default.nix (hunk 3 above)
Comment (dotlambda) [reply pkgs/development/python-modules/shapely/default.nix]: So, you don't want to use `substituteAll` in `patches`?

File: pkgs/development/python-modules/shapely/default.nix [hunk 8]
```diff
@@ -22,11 +22,16 @@ buildPythonPackage rec {
     export LANG="en_US.UTF-8";
   '';
 
-  patchPhase = let
+  postPatch =  let
     libc = if stdenv.isDarwin then "libc.dylib" else "libc.so.6";
```
Comment (dotlambda) [inline pkgs/development/python-modules/shapely/default.nix]: This could actually just be `"libc${stdenv.hostPlatform.extensions.sharedLibrary}"`.

**Comment by dotlambda** [comment]
Two further things that should be changed in your commit messages:
* "depreciated" -> "deprecated"
* pythonPackages.monty: 1.0.2 -> 1.0.4

**Comment by dotlambda** [comment]
I did the changes myself. Please have a look at the changes I made to shapely.

**Comment by costrouc** [comment]
@dotlambda looks good to me! Sorry didn't have time this morning to make the edits

**Comment by dotlambda** [comment]
Sorry didn't have time this morning to make the edits

> No problem. Thank you very much for helping out!
